frontend/src/lib/k8s/cluster.ts

```diff
@@ -164,9 +166,7 @@ export function makeKubeObject<T extends KubeObjectInterface | KubeEvent>(
     static apiList<U extends KubeObject>(
       onList: (arg: U[]) => void,
       onError?: (err: ApiError) => void,
-      opts?: {
-        namespace?: string;
-      }
```

Comment:
This was only one single namespace. With your changes, it can be also an array of strings.

Reply:
Maybe you just do something like:
`{
namespace?: string;
} | Omit<ApiListOptions, 'namespace'>`

Reply:
I set QueryParams as a member of opts. It looks like
`{ namespace?: string; queryParams?: QueryParams}`